dali/operators/reader/fits_reader_op.cc

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2020-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+
+#include "dali/core/backend_tags.h"
+#include "dali/kernels/slice/slice_cpu.h"
+#include "dali/kernels/slice/slice_flip_normalize_permute_pad_cpu.h"
+#include "dali/kernels/transpose/transpose.h"
+#include "dali/core/static_switch.h"
+#include "dali/operators/reader/fits_reader_op.h"
+#include "dali/pipeline/data/backend.h"
+
+namespace dali {
+
+static void CopyHelper(SampleView<CPUBackend> output, ConstSampleView<CPUBackend> input,
+                       ThreadPool &thread_pool, int min_blk_sz, int req_nblocks) {
+  auto *out_ptr = static_cast<uint8_t *>(output.raw_mutable_data());
+  const auto *in_ptr = static_cast<const uint8_t *>(input.raw_data());
+  auto nelements = input.shape().num_elements();
+  auto nbytes = nelements * TypeTable::GetTypeInfo(input.type()).size();
+  if (nelements <= min_blk_sz) {
+    thread_pool.AddWork([=](int tid) {
+      std::memcpy(out_ptr, in_ptr, nbytes);
+    }, nelements);
+  } else {
+    int64_t prev_b_start = 0;
+    for (int b = 0; b < req_nblocks; b++) {
+      int64_t b_start = prev_b_start;
+      int64_t b_end = prev_b_start = nbytes * (b + 1) / req_nblocks;
+      int64_t b_size = b_end - b_start;
+      thread_pool.AddWork([=](int tid) {
+        std::memcpy(out_ptr + b_start, in_ptr + b_start, b_size); 
+      }, b_size);
+    }
+  }
+}
+
+DALI_REGISTER_OPERATOR(readers__Fits, FitsReaderCPU, CPU);
+
+DALI_SCHEMA(readers__Fits)
```

Comment:
And adjust here as well:
```suggestion
DALI_SCHEMA(experimental__readers__Fits)
```

Reply:
Addressed.